### PR TITLE
Add bindings for client.readQuery

### PR DESCRIPTION
**Pull request by tatchi**
This add bindings for `client.readQuery`

### CODE REVIEW

File: src/UrqlClient.re [hunk 1]
```diff
@@ -511,6 +511,32 @@ let query =
...
+[@bs.send]
+external readQueryJs:
+  (
+    ~client: t,
+    ~query: string,
+    ~variables: Js.Json.t=?,
+    ~context: ClientTypes.partialOperationContextJs=?,
+    unit
+  ) =>
+  Js.Null.t('response) =
```
Comment (tatchi) [inline src/UrqlClient.re:523]: I was not sure between `Js.Null.t` and `Js.Nullable.t`. I chose the former since it should reflect better the return type from the js side: https://github.com/FormidableLabs/urql/blob/master/packages/core/src/client.ts#L266

Comment (parkerziegler) [reply src/UrqlClient.re:523]: ✅ I think that makes sense. AFAIK, `Js.Nullable.t` is generally preferred to handle both `null` and `undefined` but since, as you point out, the JS side explicitly uses `null` I think this looks great.

File: src/UrqlClient.re [hunk 2]
```diff
@@ -511,6 +511,32 @@ let query =
   |> Wonka.toPromise;
 };
 
+[@bs.send]
+external readQueryJs:
```
Comment (parkerziegler) [inline src/UrqlClient.re:515]: So in this case w/ `readQueryJs` you seem to be writing a binding for `readQuery` but never actually using it; instead, `readQuery` below just re-implements the same method in Reason. I actually think the latter is nicer – it takes advantage of the niceties of Reason to do what `pipe` does in `urql`.

TL;DR I think you can get rid of L514 – 524 altogether 😄

**Comment by parkerziegler** [comment]
👋 Thanks @tatchi for this, it looks really nice. The only concern I have is that we're binding to a much earlier version of `urql` currently (v1.5.1). So your `[@bs.send]` binding wouldn't actually work, b/c prior to v1.9.0 the `readQuery` method didn't exist on the client 😬 I'm still happy to keep the PR around, however, and we can merge it once we get the bindings up to speed there.

Removing the `[@bs.send]` binding as I commented would fix the binding issue, but also mean we introduce a feature earlier than expected in the bindings, which I'm very wary of doing. If you're up to making the discussed changes I'd be happy to merge this once v1.9.0 is slated for bindings.

File: src/UrqlClient.re (hunk 2 above)
Comment (tatchi) [reply src/UrqlClient.re]: Yes that makes sense. Actually I've been influenced by the `queryJs` and `mutationJs` ones. If I'm not mistaken, those are not used either. What's the reason to keep them then? Is it because one version returns a `Wonka.sourceT` while the other returns a `promise`?

**Comment by tatchi** [comment]
> 👋 Thanks @tatchi for this, it looks really nice. The only concern I have is that we're binding to a much earlier version of `urql` currently (v1.5.1). So your `[@bs.send]` binding wouldn't actually work, b/c prior to v1.9.0 the `readQuery` method didn't exist on the client 😬 I'm still happy to keep the PR around, however, and we can merge it once we get the bindings up to speed there.
> 
> Removing the `[@bs.send]` binding as I commented would fix the binding issue, but also mean we introduce a feature earlier than expected in the bindings, which I'm very wary of doing. If you're up to making the discussed changes I'd be happy to merge this once v1.9.0 is slated for bindings.

Thanks for the review! 

Yes, that totally makes sense to wait for 1.9.0 before merging this. If I understand it correctly it's probably coming soon thanks to #160

File: src/UrqlClient.re (hunk 2 above)
Comment (parkerziegler) [reply src/UrqlClient.re]: Ah that's actually a good shout, they can be removed as well! Good eye 👀

**Comment by tatchi** [comment]
Hey, thanks for the update. I'm indeed interested to finish this one 👍 I'll give it a shot in the coming days and will you know once it's done :)

**Comment by tatchi** [comment]
@parkerziegler Should be ok :)

File: src/Client.re [hunk 3]
```diff
@@ -397,3 +397,30 @@ let mutation =
...
+
+let readQuery =
+    (
+      ~client,
+      ~request,
+      ~fetchOptions=?,
+      ~requestPolicy=?,
+      ~url=?,
+      ~pollInterval=?,
+      ~meta=?,
```
Comment (parkerziegler) [inline src/Client.re:409]: One thing to note – we recently added support for the full spectrum of options that can be passed in a `Partial<OperationContext>`. These new additions include `fetch` (custom `fetch` implementation to use), `additionalTypenames` (let the operation know that it depends on specific `__typenames`, used mostly by the standard document cache), `suspense` (a boolean flag to enable suspense), and `preferGetMethod` (instruct GraphQL operations to be executed with `GET` rather than `POST` as the http method).

You can look at `executeQuery` and match that signature (since it's the same) and we should be good to go! I'm also realizing I need to update the bindings for `query` and `mutation` while I'm here as well.

**Comment by parkerziegler** [comment]
Looking great @tatchi, one last set of changes then we are good to merge 🚀

File: src/Client.re (hunk 3 above)
Comment (tatchi) [reply src/Client.re]: Thanks for the detailed explanation 😊  I indeed took inspiration from the `query` one so I missed those extra options

**Comment by tatchi** [comment]
Thanks for the review @parkerziegler. I opened #203 to update the `query` and `mutation`

**Comment by parkerziegler** [comment]
@all-contributors please add @tatchi for code.
